src/otx/v2/adapters/torch/mmengine/engine.py

```diff
@@ -40,135 +73,190 @@ class MMXEngine(Engine):
-                # print log every 100 iterations.
-                "logger": {"type": "LoggerHook", "interval": 100},
-                # enable the parameter scheduler.
-                "param_scheduler": {"type": "ParamSchedulerHook"},
-                # save checkpoint per epoch, and automatically save the best checkpoint.
-                "checkpoint": {
-                    "type": "CheckpointHook",
-                    "interval": 1,
-                    "max_keep_ckpts": 1,
-                    "save_best": "auto",
-                },
-                # set sampler seed in distributed evrionment.
-                "sampler_seed": {"type": "DistSamplerSeedHook"} if distributed else None,
-            }
-        kwargs["default_hooks"] = default_hooks
-        visualizer = get_value_from_config("visualizer", func_args, config=self.config)
-        if visualizer is not None:
-            self.config["visualizer"] = visualizer
-            if isinstance(visualizer, dict):
-                self.config["visualizer"]["_scope_"] = self.registry.name
+            runner_config["randomness"] = {"seed": seed, "deterministic": deterministic}
+
+        # Update param_scheduler, default_hooks, custom_hooks, visualizer
+        for runner_key in ("param_scheduler", "default_hooks", "custom_hooks", "visualizer"):
+            runner_value = self._get_value_from_config(runner_key, func_args)
+            if runner_value is not None:
+                runner_config[runner_key] = runner_value
 
         # Update scope for Registry import step
-        kwargs["default_scope"] = self.registry.name
+        runner_config["default_scope"] = self.registry.name
 
-        # kwargs -> Update config
-        for kwarg_key, kwarg_value in kwargs.items():
-            if kwarg_value is None:
-                continue
-            self.config[kwarg_key] = kwarg_value
         # Check Config Default is not None
         runner_default_args = get_default_args(Runner.__init__)
         for not_none_arg, default_value in runner_default_args:
-            if self.config.get(not_none_arg) is None:
-                self.config[not_none_arg] = default_value
+            if runner_config.get(not_none_arg, None) is None:
+                runner_config[not_none_arg] = default_value
         # Last Check for Runner.__init__
         runner_arg_list = get_all_args(Runner.__init__)
-        removed_key = [config_key for config_key in self.config if config_key not in runner_arg_list]
+        removed_key = [config_key for config_key in runner_config if config_key not in runner_arg_list]
         if removed_key:
             msg = f"In Engine.config, remove {removed_key} that are unavailable to the Runner."
             logger.warning(msg, stacklevel=2)
             for config_key in removed_key:
-                self.config.pop(config_key)
-        return update_check
+                runner_config.pop(config_key)
+        return runner_config, update_check
```

Comment:
I want to suggest other approach for this function. 
We now have self.default_config, func_args, kwargs. 
kwargs can be directly merge with self.default_config 
And func_args need some change to merge with self.default_config directly.
At this stage, I think only difference is randomness. So how about this kind of implementation?
`
runner_config = deepcopy(self.default_config)
_config_from_kwargs = Config(kwargs)
runner_config.merge_from_dict(_config_from_kwargs)
_config_from_func_args = Config(func_args)
randomness = {
    "seed": _config_from_func_args.pop("seed", runner_config.get("seed", 1234), 
    "deterministic": _config_from_func_args.pop("deterministic", runner_config.get("determinisitc", False)
}
_config_from_func_args.randomness = randomness
runner_config.merge_from_dict(_config_from_func_args)
`
I might forget details, so my work cannot run properly. 
Anyway could you share your opinion for this kind of implementation?

Reply:
Let me check this out.

Reply:
I checked and out function (update_config) change a total of six variables in the args for the runner.
`seed`, `deterministic`, `optimizer`, `precision`, `max_iters`, `max_epochs`
We should pop all of these variables from func_args (_config_from_func_args), modify them appropriately, and then merge them, but it may be safer to stick with the current code as it won't change the size or flow.

If you have a better idea, please let me know.